tests/pipelines/test_pipelines_common.py

```diff
@@ -2395,40 +2320,62 @@ def test_pipeline_level_group_offloading_sanity_checks(self):
+            # the layers is not traced correctly. This causes errors. For apply group offloading to VAE, a
+            # warmup forward pass (even with dummy small inputs) is recommended.
+            exclude_modules = []
+            for name, component in pipe.components.items():
+                for name in ["vae", "vqvae", "image_encoder"]:
+                    exclude_modules.append(name)
+            pipe.enable_group_offload(
+                exclude_modules=exclude_modules, onload_device=torch_device, **group_offloading_kwargs
+            )
+            for component_name, component in pipe.components.items():
+                if component_name not in exclude_modules and isinstance(component, torch.nn.Module):
+                    self.assertTrue(
+                        all(
+                            module._diffusers_hook.get_hook("group_offloading") is not None
+                            for module in component.modules()
+                            if hasattr(module, "_diffusers_hook")
+                        )
+                    )
 
-        pipe.to("cpu")
-        del pipe
+        def run_forward(pipe):
+            torch.manual_seed(0)
+            inputs = self.get_dummy_inputs(torch_device)
+            return pipe(**inputs)[0]
 
-        # Inference with offloading
-        pipe: DiffusionPipeline = self.pipeline_class(**components)
-        offload_device = "cpu"
-        pipe.enable_group_offload(
-            onload_device=torch_device,
-            offload_device=offload_device,
-            offload_type="leaf_level",
-        )
-        pipe.set_progress_bar_config(disable=None)
-        inputs["generator"] = torch.manual_seed(0)
-        out_offload = pipe(**inputs)[0]
+        pipe = create_pipe().to(torch_device)
+        output_without_group_offloading = run_forward(pipe)
 
-        max_diff = np.abs(to_np(out) - to_np(out_offload)).max()
-        self.assertLess(max_diff, expected_max_difference)
+        pipe = create_pipe()
+        enable_group_offload_on_component(pipe, {"offload_type": "block_level", "num_blocks_per_group": 1})
+        output_with_group_offloading1 = run_forward(pipe)
+
+        pipe = create_pipe()
+        enable_group_offload_on_component(pipe, {"offload_type": "leaf_level"})
+        output_with_group_offloading2 = run_forward(pipe)
+
+        if torch.is_tensor(output_without_group_offloading):
+            output_without_group_offloading = output_without_group_offloading.detach().cpu().numpy()
+            output_with_group_offloading1 = output_with_group_offloading1.detach().cpu().numpy()
+            output_with_group_offloading2 = output_with_group_offloading2.detach().cpu().numpy()
+
+        self.assertTrue(np.allclose(output_without_group_offloading, output_with_group_offloading1, atol=1e-4))
+        self.assertTrue(np.allclose(output_without_group_offloading, output_with_group_offloading2, atol=1e-4))
```

Comment:
Since the plan is to remove unittest as a dep, let's use standard asserts when adding new tests.